pool/src/storage.rs

```diff
@@ -411,6 +422,32 @@ pub fn del_queued_reserve_set(e: &Env, asset: &Address) {
     e.storage().temporary().remove(&key);
 }
 
+/// Fetch the reserve flags summary
+pub fn get_res_flags_summary(e: &Env) -> Map<Address, u32> {
+    get_persistent_default(
+        e,
+        &Symbol::new(e, RES_FLAGS_SUMMARY),
+        || map![e],
+        LEDGER_THRESHOLD_SHARED,
+        LEDGER_BUMP_SHARED,
+    )
+}
+
+/// Set the reserve flags summary
+///
+/// ### Arguments
+/// * `summary` - The map of reserve's status summary by reserve address to a status u32
+pub fn set_res_flags_summary(e: &Env, summary: &Map<Address, u32>) {
+    e.storage()
+        .persistent()
+        .set::<Symbol, Map<Address, u32>>(&Symbol::new(e, RES_FLAGS_SUMMARY), summary);
+    e.storage().persistent().extend_ttl(
+        &Symbol::new(e, RES_FLAGS_SUMMARY),
+        LEDGER_THRESHOLD_SHARED,
+        LEDGER_BUMP_SHARED,
+    );
+}
+
```

Comment:
I don't think this optimization is necessary, as we need to load `reserve_config` for each asset when conduction `gulp_emissions` anyway. Please see comment on `emissions/manager.rs` for context.

Thus, don't need to maintain a second source of truth for the asset status, and can remove the `res_flag_summary` code.